plugins/tiddlywiki/sqlite3store/init-sqlite3.js

```diff
@@ -0,0 +1,66 @@
+/*\
+title: $:/plugins/tiddlywiki/sqlite3store/init-sqlite3.js
+type: application/javascript
+
+Initialise sqlite3 and then boot TiddlyWiki
+
+This file is spliced into the HTML file to be executed after the boot kernel has been loaded.
+
+\*/
+
+(function() {
+
+// Get the main tiddler store out of the HTML file
+var storeEl = document.querySelector("script.tiddlywiki-tiddler-store"),
+	tiddlerStore = JSON.parse(storeEl.textContent);
+
+// Helper to get a tiddler from the store by title
+function getTiddler(title) {
+	for(var t=0; t<tiddlerStore.length; t++) {
+		var tiddler = tiddlerStore[t];
+		if(tiddler.title === title) {
+			return tiddler;
+		}
+	}
+	return undefined;
+}
+// Get the shadow tiddlers of this plugin
+var thisPlugin = getTiddler("$:/plugins/tiddlywiki/sqlite3store"),
+	thisPluginTiddlers = JSON.parse(thisPlugin.text).tiddlers;
+// Execute the sqlite3 module
+var sqlite3js = thisPluginTiddlers["$:/plugins/tiddlywiki/sqlite3store/sqlite3.js"].text,
+	context = {
+		exports: {}
+	};
+$tw.utils.evalSandboxed(sqlite3js,context,"$:/plugins/tiddlywiki/sqlite3store/sqlite3.js",true);
+// Create a Blob URL for the wasm data
+var sqlite3wasm = thisPluginTiddlers["$:/plugins/tiddlywiki/sqlite3store/sqlite3.wasm"].text;
+var decodedData = window.atob(sqlite3wasm),
```

Comment:
I get

```
Uncaught InvalidCharacterError: Failed to execute 'atob' on 'Window': The string to be decoded contains characters outside of the Latin1 range.
```

when mimicking your usage.

Reply:
@linonetwo, notice Jeremy's tiddlywiki.info is encoding the file as base64. The `atob` can't decode it if it wasn't encoded.

```json
		{
			"file": "sqlite3.wasm",
			"encoding": "base64",
			"fields": {
				"type": "application/wasm",
				"title": "$:/plugins/tiddlywiki/sqlite3store/sqlite3.wasm"
			}
		}
```

Reply:
@btheado I have a few tries, even I change encoding to utf8, when I use `$tw.wiki.getTiddlerText`, I always get the base64 encoded test. Same even set `"type": "text/text"`.

I tried this because 30MB wasm become 80MB of string after become a tiddler and I want to reduce it.

Reply:
To be clear, TiddlyWiki doesn't currently support true binary tiddlers; we only support base64 encoded pseudo-binary tiddlers.